### PR TITLE
refactor: small refactor in threads for waiting Flow readiness

### PR DESCRIPTION
**Pull request by JoanFM**
**Goals:**
Task from rich local in the thread stack

### CODE REVIEW

File: jina/helper.py [hunk 1]
```diff
@@ -1237,15 +1237,8 @@ def _get_ip(url):
...
-    threads = []
-
     for idx, ip in enumerate(ip_server_list):
-        t = threading.Thread(target=_get_ip, args=(ip,))
-        threads.append(t)
-        t.start()
-
-    for t in threads:
-        t.join(timeout)
+        results.append(_get_ip(ip))
```
Comment (samsja) [inline jina/helper.py:1241]:
```suggestion
        r = _get_ip(ip)
        if r:
            return r
```

Now that it is not async we can just return the first one that is not None to save time. (As we only return one IP later anyway)

File: jina/helper.py [hunk 2]
```diff
@@ -1237,15 +1237,8 @@ def _get_ip(url):
...
 
     for r in results:
```
Comment (samsja) [inline jina/helper.py:1243]: this can be removed if we apply the change in the for loop

File: jina/helper.py [hunk 3]
```diff
@@ -1237,15 +1237,8 @@ def _get_ip(url):
...
         if r:
```
Comment (samsja) [inline jina/helper.py:1244]: this can be removed if we apply the suggestion above

File: jina/helper.py (hunk 1 above)
Comment (JoanFM) [reply jina/helper.py:1241]: Good

File: jina/helper.py (hunk 3 above)
Comment (JoanFM) [reply jina/helper.py]: Suggestion applied, good catch!

**Review by girishc13 — APPROVED** [review]

**Review by samsja — APPROVED** [review]

**Pull request merged**